ocaml/testsuite/tests/typing-modes/crossing.ml

```diff
@@ -2,15 +2,60 @@
  expect;
 *)
 
+(* mode crossing during inclusion check is according to the written type, not
+the inferred type. *)
+
+(* In this example, the inferred type does not allow mode crossing, but the
+written type does. *)
 module M : sig
     val f : int @ nonportable -> int @ portable
 end = struct
-    let f (_ @ portable) = (42 : _ @@ nonportable)
+    let f (x @ portable) = (x : _ @@ nonportable)
 end
 [%%expect{|
 module M : sig val f : int -> int @ portable end
 |}]
 
+(* In this example, the inferred type allows crossing to portable, but the
```

Comment:
Rejecting this is a bit unfortunate, because it would be sound to accept. When the inferred/written types differ, mode crossing at *either one* is fine, essentially because the mode-cross and subtype operations can compose in either direction. I can even imagine scenarios where we want mode-cross then subtype then mode-cross — which would be fine! (Maybe that doesn’t exist for the current subtype relation; not sure.)

I don’t think we need to implement the mode-cross-at-either-type behavior right now, but maybe just leave a comment here saying that we could do better. If folks complain, we’ll return.